pyfar/dsp/dsp_interpolation.py

```diff
@@ -0,0 +1,731 @@
+import numpy as np
+from scipy.special import iv as bessel_first_mod
+from scipy.interpolate import interp1d
+import scipy.signal as sgn
+import matplotlib.pyplot as plt
+import pyfar as pf
+
+
+def smooth_fractional_octave(signal, width, mode="magnitude",
+                             window="boxcar"):
+    """
+    Smooth spectrum with a fractional octave width.
+
+    The smoothing is done according to Tylka et al. 2017 [#]_ method 2 in three
+    steps:
+
+    1. Interpolate the spectrum to a logarithmically spaced frequency scale
+    2. Smooth the spectrum by convolution with a smoothing window
+    3. Interpolate the spectrum to the original linear frequency scale
+
+    Note: Method 3 in Tylka at al. 2017 is mathematically more elegant at the
+    price of a largely increased computational and memory cost. In most
+    practical cases, methods 2 and 3 yield close to identical results (cf. Fig.
+    2 and 3 in Tylka et al. 2017). If the spectrum contains extreme
+    discontinuities, however, method 3 is superior (see examples below).
+
+    Parameters
+    ----------
+    signal : pyfar.Signal
+        The input data.
+    width : number
+        The width of the smoothing window in octaves, e.g., 1/3 will apply
+        third octave smoothing.
```

Comment:
I think it would be nice to be consistent with the fractional octave functions in dsp.filter here:
`num_fractions (int) – The number of bands an octave is divided into. Eg., 1 refers to octave bands and 3 to third octave bands.`

Reply:
I also thought of this. But the filters split bands return 3 bands per octave for example, whereas smoothing applies a window of a certain width. That is why I used different names.

Reply:
Hm, I'd still slightly prefer consistency here, as from my perspecitve the meaning of the parameters is very similar. Let's discuss with a second reviewer :)

Reply:
Renamed it as suggested, but still allow non-integer numbers. Although this might not haven often, it is possible and I think we should not restrict it.